formtools/wizard/views.py

```diff
@@ -254,34 +255,52 @@ def dispatch(self, request, *args, **kwargs):
     def get(self, request, *args, **kwargs):
         """
         This method handles GET requests.
-
-        If a GET request reaches this point, the wizard assumes that the user
-        just starts at the first step or wants to restart the process.
-        The data of the wizard will be resetted before rendering the first step
         """
+
+        if self.form_method == 'GET':
+            current_step = self.request.GET.get('{}-current_step'.format(self.prefix), None)
+
+            if current_step is None:
+                # If a GET request reaches this point, the wizard assumes that the user
+                # just starts at the first step or wants to restart the process.
+                # The data of the wizard will be resetted before rendering the first step
+                self.storage.reset()
+
+                # reset the current step to the first step.
+                self.storage.current_step = self.steps.first
+
+                return self.render(self.get_form())
+
+            return self.handle_step(*args, **kwargs)
+
         self.storage.reset()
 
         # reset the current step to the first step.
         self.storage.current_step = self.steps.first
+
         return self.render(self.get_form())
 
     def post(self, *args, **kwargs):
         """
         This method handles POST requests.
+        """
+        return self.handle_step(*args, **kwargs)
 
+    def handle_step(self, *args, **kwargs):
+        """
         The wizard will render either the current step (if form validation
         wasn't successful), the next step (if the current step was stored
         successful) or the done view (if no more steps are available)
         """
         # Look for a wizard_goto_step element in the posted data which
         # contains a valid step name. If one was found, render the requested
         # form. (This makes stepping back a lot easier).
-        wizard_goto_step = self.request.POST.get('wizard_goto_step', None)
+        wizard_goto_step = getattr(self.request, self.form_method).get('wizard_goto_step', None)
```

Comment:
I'm a bit confused as to how this is supposed to work. By ignoring the actual `request.method` and forcing the use of `self.form_method`, isn't the code saying that ALL form interaction can only use self.form_method? Specifically, stepping forward and the final submit cannot use a POST if self.form_method is set to 'GET'.

Or have I not understood the purpose of self.form_method?